### PR TITLE
chore(toolchain): Fix lint and test for beta toolchain

### PR DESCRIPTION
**Pull request by olksdr**
Make sure we also prepare for the next Rust release. 

This PR fixes lint for beta toolchain, and make sure all the tests are still passing.

#skip-changelog

### CODE REVIEW

**Review by jjbayer — APPROVED** [review]
Nice! Do we also want to run this in CI, as a non-required check?

**Comment by olksdr** [comment]
> Nice! Do we also want to run this in CI, as a non-required check?

We do have this in CI https://github.com/getsentry/relay/actions/workflows/beta.yml and it runs periodically.  

Maybe adding this to our common CI runs will bring more benefits.  But also sometimes `beta` has something which `stable` does not support yet, and it is just not possible to fix it on both at the same time, so we will be having 1 red job.

File: relay-general/src/types/meta.rs [hunk 1]
```diff
@@ -474,7 +474,7 @@ impl Meta {
     }
 
     fn upsert(&mut self) -> &mut MetaInner {
-        self.0.get_or_insert_with(|| Box::new(MetaInner::default()))
+        self.0.get_or_insert_with(|| Box::<MetaInner>::default())
```
Comment (jan-auer) [inline relay-general/src/types/meta.rs]: This should work without type hints.

```suggestion
        self.0.get_or_insert_with(|| Box::default())
```

**Review by jan-auer — APPROVED** [review]

File: relay-general/src/types/meta.rs (hunk 1 above)
Comment (olksdr) [reply relay-general/src/types/meta.rs]: I haven't tried it, just used what clippy suggested. Will try it now and make a change.

**Pull request merged**